tools/scoring/score.py

```diff
@@ -265,86 +318,77 @@ def score(division,
+    else:
+        data_new['score_modifier'] = 1
 
+    data_solved = data_new[(data_new.result.isin(set(verdicts)))]
     if g_args.sequential:
-        data = data[(data.cpu_time <= wclock_limit)]
+        data_solved = data_solved[(data_solved.cpu_time <= wclock_limit)]
     else:
-        data = data[(data.wallclock_time <= wclock_limit)]
+        data_solved = data_solved[(data_solved.wallclock_time <= wclock_limit)]
 
     if skip_unknowns:
-        data = data[(data.result == data.expected)]
+        data_solved = data_solved[(data_solved.result == data_solved.expected)]
     else:
-        data = data[(data.expected == "starexec-unknown")
-                    | (data.result == data.expected)]
-
-    for solver, sdata in data.groupby('solver'):
-        if g_args.log: log("Computing scores for "+solver)
-        nsolved = len(sdata)
-        error = 0.0
-        correct = 0.0
-        wall_total = 0.0
-        cpu_total = 0.0
-        wrong_solver = all_wrong[all_wrong.solver == solver]
-        for family, sfdata in sdata.groupby('family'):
-            modifier = family_scores[family] if use_families else 1
-
-            sf_wrong = len(wrong_solver[wrong_solver.family == family])
-            sf_solved = len(sfdata)
-
-            this_mod = benchmark_num * modifier if use_families else 1
-
-            error += sf_wrong * this_mod
-            correct += sf_solved * this_mod
-            wall_total += sfdata.wallclock_time.sum() * modifier
-            cpu_total += sfdata.cpu_time.sum() * modifier
-            #if g_args.log: log("... {}, {}, {}, {} with {}".format(\
-            #       error, correct, wall_total, cpu_total, modifier))
-
-        psolved = 100.0 * (float(nsolved) / benchmark_num)
-        assert psolved > 0
-        row = [year, division, solver, psolved, error, correct,
-               wall_total, cpu_total]
-        rows.append(row)
-        if g_args.log: log("Row: {}".format(row))
-
-    # After computing a row per solver we then sort and rank them
-    # Note that competitive solvers cannot get awarded a rank and merely
-    # get the current rank without increasing it
-    rows.sort(key=row_key)
-    rank = 0
-    for row in rows:
-        row.append(rank)
-        competitive = competitive_row(row)
-        row.append(competitive)
-        if competitive:
-            rank+=1
-    return rows
+        data_solved = data_solved[(data_solved.expected == "starexec-unknown")
+                                  | (data_solved.result == data_solved.expected)]
+
+    data_new.loc[data_solved.index, 'correct'] = 1
+
+    # Compute scores
+    data_new.score_correct = data_new.correct * data_new.score_modifier
+    data_new.score_error = data_new.error * data_new.score_modifier
+    data_new.score_cpu_time = data_new.cpu_time * data_new.alpha_prime_b
+    data_new.score_wallclock_time = \
+        data_new.wallclock_time * data_new.alpha_prime_b
+
+    # Compute time scores only for correctly solved
```

Comment:
We might want to introduce a penalty (= time limit) for memout since we rank by time in case of two matching correctness scores. And yes, we want to compute the time scores for all benchmarks, not only for correctly solved.